### PR TITLE
refact(core): update dependencies version &  adopt the new analyzer

### PR DESCRIPTION
**Pull request by jadepeng**
And improve analyzer code

### CODE REVIEW

File: hugegraph-core/src/main/java/com/baidu/hugegraph/analyzer/AnalyzerFactory.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
 public class AnalyzerFactory {
 
-    private static Map<String, Class<? extends Analyzer>> analyzers;
+    private static final Map<String, Class<? extends Analyzer>> analyzers;
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/analyzer/AnalyzerFactory.java:30]: can you split changes (like the code improve) into a separate pr?

File: hugegraph-core/pom.xml [hunk 2]
```diff
@@ -13,6 +13,22 @@
...
+        <jcseg.version>2.6.2</jcseg.version>
+        <hanlp.version>portable-1.8.3</hanlp.version>
+        <ansj-seg.version>5.1.6</ansj-seg.version>
+        <lucene-core.version>8.11.2</lucene-core.version>
+        <jieba-analysis.version>1.0.2</jieba-analysis.version>
+        <ikanalyzer.version>2012_u6</ikanalyzer.version>
+        <commons-compress.version>1.21</commons-compress.version>
+        <eclipse-collections.version>11.1.0</eclipse-collections.version>
+        <fastutil.version>8.5.9</fastutil.version>
+        <jjwt.version>0.11.5</jjwt.version>
```
Comment (javeme) [inline hugegraph-core/pom.xml:31]: can you highlight which versions are changed? or it's ok to just leave some comments in the reply

**Review by javeme — COMMENTED** [review]
Thanks for your contribution~

File: hugegraph-core/pom.xml (hunk 2 above)
Comment (jadepeng) [reply hugegraph-core/pom.xml:30]: jraft 1.3.9->1.3.11
ohc-core 0.7.0->0.7.4
org.apdplat.word 1.3->1.3.1
hanlp portable-1.5.0->portable-1.8.3
lucene-analyzers-smartcn,lucene-core 7.4.0->8.11.2
jcseg-core->2.2.0 ->2.6.2
lz4-java 1.7.1->1.8.0
eclipse-collections 10.4.0->11.1.0
fastutil 8.1.0->8.4.0
jjwt 0.11.2->0.11.5

File: hugegraph-core/src/main/java/com/baidu/hugegraph/analyzer/JcsegAnalyzer.java [hunk 3]
```diff
@@ -45,29 +44,35 @@ public class JcsegAnalyzer implements Analyzer {
...
 
     public JcsegAnalyzer(String mode) {
         if (!SUPPORT_MODES.contains(mode)) {
             throw new ConfigException(
                       "Unsupported segment mode '%s' for jcseg analyzer, " +
                       "the available values are %s", mode, SUPPORT_MODES);
         }
-        this.segMode = SUPPORT_MODES.indexOf(mode) + 1;
+
+        if (mode.equals("Simple")) {
```
Comment (imbajin) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/analyzer/JcsegAnalyzer.java:60]: seems we should revert it `"Simple".equals(more)`

File: hugegraph-core/src/main/java/com/baidu/hugegraph/analyzer/JcsegAnalyzer.java [hunk 4]
```diff
@@ -45,29 +44,35 @@ public class JcsegAnalyzer implements Analyzer {
             "Complex"
     );
 
-    private static final JcsegTaskConfig CONFIG = new JcsegTaskConfig();
+    private static final SegmenterConfig CONFIG = new SegmenterConfig();
     private static final ADictionary DIC =
-            DictionaryFactory.createDefaultDictionary(new JcsegTaskConfig());
+        DictionaryFactory.createDefaultDictionary(CONFIG);
```
Comment (imbajin) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/analyzer/JcsegAnalyzer.java:49]: If this line does not exceed 100 letters, consider merging into one line (ignore the comment  if > 100)

File: hugegraph-core/src/main/java/com/baidu/hugegraph/analyzer/JcsegAnalyzer.java [hunk 5]
```diff
@@ -45,29 +44,35 @@ public class JcsegAnalyzer implements Analyzer {
...
         try {
-            Object[] args = new Object[]{new StringReader(text), CONFIG, DIC};
-            ISegment seg = SegmentFactory.createJcseg(this.segMode, args);
-            IWord word = null;
-            while ((word = seg.next()) != null) {
+            ISegment segmentor = this.type.factory.create(CONFIG, DIC);
+            segmentor.reset(new StringReader(text));
+
+            IWord word;
+            while ((word = segmentor.next()) != null) {
```
Comment (imbajin) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/analyzer/JcsegAnalyzer.java:75]: warn: will the `segmentor.next()` throws exception if null?

Comment (jadepeng) [reply hugegraph-core/src/main/java/com/baidu/hugegraph/analyzer/JcsegAnalyzer.java:75]: No exception

**Review by imbajin — APPROVED** [review]

**Pull request merged**